src/main/java/net/coderbot/iris/shaderpack/PropertiesPreprocessor.java

```diff
@@ -0,0 +1,140 @@
+package net.coderbot.iris.shaderpack;
+
+import com.google.common.collect.ImmutableMap;
+import net.coderbot.iris.Iris;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static net.coderbot.iris.gl.shader.StandardMacros.getMcVersion;
+
+public class PropertiesPreprocessor {
+	private static final ImmutableMap<String, String> MACRO_CONSTANTS = ImmutableMap.of("MC_VERSION", getMcVersion());
+	
+	public static List<String> parseProperties(String fileName, String fileContents) {
+		List<String> lines = new ArrayList<>();
+
+		boolean currentlyParsingConditionalProperties = false;
```

Comment:
Should have a more descriptive name for this... perhaps `currentConditionPassed` or something?